slitlessutils/core/preprocess/astrometry/downgrade_wcs.py

```diff
@@ -1,19 +1,96 @@
-import os
-
 from astropy.io import fits
 
 from ....logger import LOGGER
+from . import utils
+
+
+def _downgrade_wcs_one(imgfile, key, mode, newfile, inplace):
+    if not isinstance(imgfile, str):
+        LOGGER.warning('imgfile must be a string')
+        return None
+
+    LOGGER.info(f'downgrading WCS in {imgfile} to WCSNAME{key}')
+    with fits.open(imgfile, mode=mode) as hdul:
 
+        # do some error checking... On Second thought, we probably don't
+        # need/want this.
+        # obstype = hdul[0].header['OBSTYPE']
+        # if obstype != 'IMAGING':
+        #    LOGGER.warning(f'direct image obstype ({obstype}) is not imaging.')
+        #    return
 
-def downgrade_wcs(imgfile, key='A', newfile=None, inplace=False):
+        for ext, hdu in enumerate(hdul):
+
+            wcsname = hdu.header.get('WCSNAME')
+
+            # do we check for Gaia?  Prolly not.
+            # if wcsname and (wcsname.lower().find('gaia') != -1):
+
+            # get the current WCS
+            crval = utils.get_crval(hdu.header, '')
+            cd = utils.get_cd(hdu.header, '')
+            # crvals = [hdu.header['CRVAL1'], hdu.header['CRVAL2']]
+            # cd = [[hdu.header['CD1_1'], hdu.header['CD1_2']],
+            #       [hdu.header['CD2_1'], hdu.header['CD2_2']]]
```

Comment:
Is there a reason to keep these commented out code blocks? Seems like this is the stuff that was moved to the utils functions and can be removed.

Reply:
Yeah, I realized that i was doing the same thing multiple times, so I moved them to utils.  I can clean out the garbage, but I only know how to do that in the terminal (not on the web UI).  Can you clean that fat, or want me to?